src/lib.rs

```diff
@@ -146,6 +146,86 @@
 //! # }
 //! ```
 //!
+//! # `no-std` support
+//!
+//! Serde CBOR supports building in a `no_std` context, use the following lines
+//! in your `Cargo.toml` dependencies:
+//! ``` toml
+//! [dependencies]
+//! serde = { version = "1.0", default-features = false }
+//! serde_cbor = { version = "0.10", default-features = false }
+//! ```
+//!
+//! *Note*: to use derive macros in serde you will need to declare `serde`
+//! dependency like so:
+//! ``` toml
+//! serde = { version = "1.0", default-features = false, features = ["derive"] }
+//! ```
+//!
+//! Serialize an object.
+//! ``` rust
+//! # #[macro_use] extern crate serde_derive;
+//! use serde::Serialize;
+//! use serde_cbor::Serializer;
+//! use serde_cbor::ser::SliceWrite;
+//!
+//! #[derive(Serialize)]
+//! struct User {
+//!     user_id: u32,
+//!     password_hash: [u8; 4],
+//! }
+//!
+//! let mut buf = [0u8; 100];
+//! let writer = SliceWrite::new(&mut buf[..]);
+//! let mut ser = Serializer::new(writer);
+//! let user = User {
+//!     user_id: 42,
+//!     password_hash: [1, 2, 3, 4],
+//! };
+//! user.serialize(&mut ser).unwrap();
```

Comment:
One minor note, can you change the example, so that unwrap isn't used? Thanks.

[Examples use ?, not try!, not unwrap](https://rust-lang-nursery.github.io/api-guidelines/documentation.html#examples-use--not-try-not-unwrap-c-question-mark)

Reply:
the doc example are expected to return Unit, and not a result, which results in this error:
```

---- src/lib.rs -  (line 166) stdout ----
error[E0277]: the `?` operator can only be used in a function that returns `Result` or `Option` (or another type that implements `std::ops::Try`)
  --> src/lib.rs:185:1
   |
22 | user.serialize(&mut ser)?;
   | ^^^^^^^^^^^^^^^^^^^^^^^^^ cannot use the `?` operator in a function that returns `()`
   |
   = help: the trait `std::ops::Try` is not implemented for `()`
   = note: required by `std::ops::Try::from_error`
```

I can put that in a function, but that makes the example less readable in my opinion.

Reply:
Have a look at the linked doc. You can hide the `fn main() -> Result<...>` in the HTML docs by adding a # to the start of the line.

Reply:
Ha, sorry, my mistake.